pkg/datapath/iptables/iptables.go

```diff
@@ -1217,7 +1219,77 @@ func (m *IptablesManager) installMasqueradeRules(prog iptablesInterface, ifName,
 		}
 	}
 
-	// Masquerade all egress traffic leaving the node
+	// Masquerade egress traffic leaving the node based on source routing
+	//
+	// If this option is enabled, then it takes precedence over the catch-all
+	// MASQUERADE further below.
+	if option.Config.EnableMasqueradeRouteSource {
+		devices := option.Config.GetDevices()
```

Comment:
Do we actually want to fallback to `--devices` if `--egress-masquerade-interfaces` is not set? Asking, as AFAIK `--devices` are only used to control the NodePort BPF and the BPF masq, while previously  the iptables-based MASQ didn't consider them.

Reply:
I'm thinking of adding the c.EnableMasqueradeRouteSource dependency here:
```
func (c *DaemonConfig) AreDevicesRequired() bool {
        return c.EnableNodePort || c.EnableHostFirewall || c.EnableBandwidthManager ||
                c.EnableWireguard || c.EnableHighScaleIPcache || c.EnableL2Announcements
}
```
The source selection can be tied to devices unless next hop groups are used, so both yes & no.

Reply:
Implemented the above now.